packages/docusaurus-plugin-content-docs/src/sidebars/processor.ts

```diff
@@ -98,6 +101,12 @@ async function processSidebar(
     if (item.type === 'autogenerated') {
       return processAutoGeneratedItem(item);
     }
+    if (
+      item.type === 'doc' &&
+      docIsInvisible(docs.find((doc) => doc.id === item.id))
```

Comment:
iterating on all docs for each processed doc might not be very efficient

We have a `createDocsByIdIndex(docs)` that you can use to index docs by id ahead of time